### PR TITLE
Minimal filter changes

### PR DESCRIPTION
**Pull request by kiprot**
- remove extra spacings
- add a null-check in AuthFilter
- remove unused imports 
- make class-wide variables private

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/AuthFilter.java [hunk 1]
```diff
@@ -13,68 +13,65 @@
...
+    private FirebaseApp firebaseApp;
 
     // Create an instance of FirebaseApp.
     public AuthFilter() throws IOException {
         firebaseAppInit = new FirebaseAppInit();
         firebaseApp = firebaseAppInit.initializeFirebaseApp();
     }
 
-    public void init(FilterConfig config) {}
+    public void init(FilterConfig config) throws ServletException {}
```
Comment (octopus1000) [inline src/main/java/com/google/sps/servlets/AuthFilter.java:52]: This function doesn't do anything, does it need to throw exception?

File: src/main/java/com/google/sps/servlets/AuthFilter.java [hunk 2]
```diff
@@ -13,68 +13,65 @@
...
         }
 
+        // Attach uid to header if it is not null, else send 401 error message to user.
         String uid = decodedUseridToken.getUid();
-
-        MutableHttpServletRequest mutableRequest = new MutableHttpServletRequest(req);
-
-        mutableRequest.putHeader("userId", uid);
-        chain.doFilter(mutableRequest, response);
+        if(uid != null) {
```
Comment (octopus1000) [inline src/main/java/com/google/sps/servlets/AuthFilter.java:70]: Prefer to throw error first

if (uid == null) {
  setError..
  return
}
// Do the rest stuff.

File: src/main/java/com/google/sps/servlets/AuthFilter.java (hunk 1 above)
Comment (kiprot) [reply src/main/java/com/google/sps/servlets/AuthFilter.java:52]: no. i will correct it

File: src/main/java/com/google/sps/servlets/AuthFilter.java (hunk 2 above)
Comment (kiprot) [reply src/main/java/com/google/sps/servlets/AuthFilter.java:70]: okay, noted

**Pull request merged**